bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataGrpcClient.java

```diff
@@ -403,9 +403,10 @@ public int available() {
   public ResultScanner<FlatRow> readFlatRows(ReadRowsRequest request) {
     // Delegate all resumable operations to the scanner. It will request a non-resumable scanner
     // during operation.
-    ResponseQueueReader reader = new ResponseQueueReader(retryOptions.getStreamingBufferSize());
-    final RetryingReadRowsOperation listener = createReadRowsRetryListener(request, reader);
-    return new ResumingStreamingResultScanner(reader, listener);
+    ResponseQueueReader reader = new ResponseQueueReader();
+    ScanHandler scanHandler = createReadRowsRetryListener(request, reader);
+    reader.setScanHandler(scanHandler);
```

Comment:
To avoid having both the scanner & reader calling out to the ScanHandler(RetryingReadRowsOperation) can the scanner just call the reader and the reader delegates to the ScanHandler?